src/versions/validation/event.js

```diff
@@ -10,7 +10,8 @@ const orderByFields = [
 
 const schemas = {}
 
-const getObjectSchema = (name) => Joi.string().regex(new RegExp(`^${name}\\.\\w+$`))
+// match string value like 'a.b.c' or 'a[0].b.c[2]'
+const getObjectSchema = (name) => Joi.string().regex(new RegExp(`^${name}((\\[(\\d+)\\]|\\.\\w+)*)$`))
```

Comment:
Given the complex regex it would be nice to add some pattern [name](https://hapi.dev/family/joi/?v=16.1.7#stringpatternregex-name--options---aliases-regex) to help end user.

```js
Joi.string().pattern(new RegExp(`^${name}((\\[(\\d+)\\]|\\.\\w+)*)$`), {
  name: `accessor string like ${name}.nested.arr[1]`
})
```

Reply:
Thanks!